FirebaseMessaging/Sources/FIRMessagingAnalytics.m

```diff
@@ -212,11 +216,34 @@ + (void)logMessage:(NSDictionary *)notification
       break;
 
     default:
-      // Only a silent notification (i.e. 'content-available' is true) can be received while the app
-      // is in the background. These messages aren't loggable anyway.
+      // App was either in background state or in transition from closed
+      // to open.
+      // Needs to check notification is displayed.
+      if ([[self class] isDisplayNotification:notification]) {
+        [self logOpenNotification:notification toAnalytics:analytics];
+      }
```

Comment:
Would it be possible to add tests for the new logic to guard against regressions in the future?

Reply:
I've added unit tests for isDisplayNotifications. 
Btw we have not changed this code for a long time, so I don't know if UIApplicationState has changed because I remember we used to be able to log with UIApplicationStateInactive state. We can mock the UIApplicationState, but for this scenario, we can't prevent regressions if system has some weird shift between these UIApplicationStates, but yeah feel free to add more tests if you see fit!

Reply:
I will add in another PR to ensure we release in time.